orte/orted/orted_submit.c

```diff
@@ -813,7 +813,19 @@ int orte_submit_job(char *argv[], int *index,
     }
     /* if we were asked to output to files, pass it along */
     if (NULL != orte_cmd_options.output_filename) {
-        orte_set_attribute(&jdata->attributes, ORTE_JOB_OUTPUT_TO_FILE, ORTE_ATTR_GLOBAL, orte_cmd_options.output_filename, OPAL_STRING);
+        /* if the given filename isn't an absolute path, then
+         * convert it to one so the name will be relative to
+         * the directory where prun was given as that is what
+         * the user will have seen */
```

Comment:
Does this assume that the filesystem is the same on all remote nodes?  If so, do we want to make that assumption?

Reply:
If you don't, then the file is going to go somewhere very unexpected - as reported by the user. It will wind up relative to wherever the orted defaults to going, which will vary by system/launcher.

This way, it goes relative to where the user types "mpirun", which is what the user sees and knows.

Reply:
Just to clarify: is it relative to the PWD of `mpirun`, or is it relative to the app working directory on each node?

Reply:
@rhc54 and I talked on the phone: we rationalized that this solution is "good enough" -- we're just documenting in the mpirun help output and the man page exactly what it does.